drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -134,6 +143,38 @@ static int lis2dh_fetch_xyz(const struct device *dev,
 	struct lis2dh_data *lis2dh = dev->data;
 	int status = -ENODATA;
 	size_t i;
+
+#if defined(CONFIG_LIS2DH_FIFO_MODE)
+	static uint8_t buf[LIS2DH_FIFO_SZ];
+	uint8_t fifo_src, bytes_in_fifo;
+
+	status = lis2dh->hw_tf->read_reg(dev, LIS2DH_FIFO_SRC_REG, &fifo_src);
+	if (status < 0) {
+		LOG_WRN("Could not read FIFO_SRC reg");
+		return status;
+	}
+
+	if (fifo_src & LIS2DH_FIFO_SRC_EMPTY) {
+		return -ENODATA;
+	}
+
+	bytes_in_fifo = ((fifo_src & LIS2DH_FIFO_SRC_N_MASK) + 1) * 6;
+	status = lis2dh->hw_tf->read_data(dev, LIS2DH_REG_ACCEL_X_LSB,
+					  buf, bytes_in_fifo);
+	if (status < 0) {
+		LOG_WRN("Could not read accel axis data");
+		return status;
+	}
+
+	status = ring_buf_put(&lis2dh->fifo_rb, buf, bytes_in_fifo);
```

Comment:
This one is copying data, right?
So, from the golobal buf[] to the per-instance private fifo_rb.
Is there any way to avoid the copy?

Reply:
Possibly, this could use put_claim/put_finish to get a buffer pointer to write into with ring_buf, but it may not be the size you need in a single shot. In which case you'd have to handle the wrap around.